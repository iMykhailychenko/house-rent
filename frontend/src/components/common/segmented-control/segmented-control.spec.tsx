import { fireEvent, render } from '@testing-library/react';

import SegmentedControl from './segmented-control';

const mockSegments = [
    { id: 'test', name: 'test' },
    { id: 'test1', name: 'test1' },
    { id: 'test2', name: 'test2' },
];

describe('SegmentedControl component', () => {
    it('matches snapshot', () => {
        const handleChange = jest.fn();
        const { asFragment } = render(<SegmentedControl value={mockSegments} active="test" onChange={handleChange} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('toggle active segment', () => {
        const handleChange = jest.fn();
        const { getByTitle, getByText } = render(<SegmentedControl value={mockSegments} active="test" onChange={handleChange} />);
        expect(getByTitle('test').classList.contains('active')).toBeTruthy();

        fireEvent.click(getByText('test2'));
        expect(handleChange).toHaveBeenCalledWith('test2');
    });
});
