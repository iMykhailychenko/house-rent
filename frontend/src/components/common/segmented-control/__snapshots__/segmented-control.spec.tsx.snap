// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControl component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <ul
      class="list"
    >
      <li
        class="item active"
        title="test"
      >
        <button
          type="button"
        >
          test
        </button>
      </li>
      <li
        class="item"
        title="test1"
      >
        <button
          type="button"
        >
          test1
        </button>
      </li>
      <li
        class="item"
        title="test2"
      >
        <button
          type="button"
        >
          test2
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
